src/scss/_date.scss

```diff
@@ -0,0 +1,57 @@
+// sass-lint:disable no-qualifying-elements
+/// @access private
+/// @param {Boolean} $disabled Whether to output disabled state styling
+/// @param {Boolean} $invalid Whether to output invalid state styling
+/// @param {Boolean} $valid Whether to output valid state styling
+/// @output Styling for date inputs
+@mixin _oFormsDate (
+	$disabled: null,
+	$invalid: null,
+	$valid: null
+) {
+	$element: 'input[type=text]';
+
+	.o-forms-input--date {
+		clear: both;
+
+		label {
+			@include oTypographyMargin($bottom: 0);
+			float: left;
+		}
+
+		@if $invalid {
+			&.o-forms-input--invalid {
+				@include _oFormsInputValidity('invalid', $element);
```

Comment:
Would it be possible to output some of these base styles at the top level, inside the `oForms` mixin? Right now we're duplicating a load of output CSS for each input type.

If we have a top-level `.o-forms-input--invalid` selector which this then expands on, then we'd reduce quite a lot I think. The same goes for `.o-forms-input--valid`

Reply:
Turns out we rely in on that class to apply validity to an input over the `:invalid` / `:valid` pseudo-selectors.

There's no straightforward way (as far as I have seen) of adding those pseudo selectors to the markup, and the browser engine will apply them automatically. E.g. if an input is `:required`, all required inputs would load as invalid until they are filled out.

We also need the individual styles to be specific about which element to  affect under what condition—otherwise the styles could apply to anything under the `--invalid` or `--valid` parent class, as they are broader, such as `color` or `background-color`

Reply:
Having said that, I could modify `input`, `select` and `textarea` and remove the `$invalid`, `$valid` options permanently 🤔